### PR TITLE
Release 6.9.0

### PR DESCRIPTION
**Pull request by crow**
Manually tested on iOS and Android devices

Able to receive tag push, rich push and run ^+t action

Manually tested the opt in event on iPad by breaking on the point where the event is passed to the bridge and checking the event contents. 

Procedure: 
1. Opted in
2. Opted out in system settings
3. Re-opened the app
4. Ensured the breakpoint was hit
5. Checked the notification contents matched up with the notification alert types changed in system settings

### CODE REVIEW

File: www/UrbanAirship.js [hunk 1]
```diff
@@ -183,6 +183,18 @@ module.exports = {
    * @param {number} [notification_id] The Android notification ID.
    */
 
+   /**
+    * Event fired when the user notification opt-in status changes.
```
Comment (rlepinski) [inline www/UrbanAirship.js]: do we need to add this to Android?

File: src/android/UAirshipPlugin.java [hunk 2]
```diff
@@ -113,16 +105,14 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
...
         Autopilot.automaticTakeOff(cordova.getActivity().getApplication());
+
     }
 
     @Override
     public void onResume(boolean multitasking) {
         super.onResume(multitasking);
 
-        // Handle any Google Play services errors
-        if (PlayServicesUtils.isGooglePlayStoreAvailable(cordova.getActivity())) {
```
Comment (rlepinski) [inline src/android/UAirshipPlugin.java]: do we want to remove this?

File: src/android/UAirshipPlugin.java [hunk 3]
```diff
@@ -1032,10 +1022,10 @@ void displayInboxMessage(JSONArray data, CallbackContext callbackContext) throws
             @Override
             public void run() {
                 Intent intent = new Intent(cordova.getActivity(), CustomMessageActivity.class)
-                        .setAction(RichPushInbox.VIEW_MESSAGE_INTENT_ACTION)
-                        .setPackage(cordova.getActivity().getPackageName())
-                        .setData(Uri.fromParts(RichPushInbox.MESSAGE_DATA_SCHEME, messageId, null))
-                        .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_SINGLE_TOP);
```
Comment (rlepinski) [inline src/android/UAirshipPlugin.java]: Whats with all the whitespace changes?

File: src/android/UAirshipPluginManager.java [hunk 4]
```diff
@@ -102,6 +111,28 @@ void deepLinkReceived(String deepLink) {
...
 
     /**
+     * Called on app resume and when registration changes.
+     *
+     * @param context The application context.
+     */
+    void checkOptInStatus(Context context) {
+        boolean optIn = UAirship.shared().getPushManager().isOptIn();
+
+        SharedPreferences preferences = context.getSharedPreferences(OPT_IN_STATUS_SHARED_PREFERENCES_FILE, Context.MODE_PRIVATE);
```
Comment (rlepinski) [inline src/android/UAirshipPluginManager.java]: you should cache this

**Comment by rlepinski** [comment]
Missing a change in the airship receiver?

File: src/android/build-extras.gradle [hunk 5]
```diff
@@ -14,5 +14,5 @@ dependencies {
     compile 'com.google.android.gms:play-services-location:[11.0.1,)'
     compile 'com.android.support:appcompat-v7:25.3.1'
 
-    compile 'com.urbanairship.android:urbanairship-sdk:8.6.0'
+    compile 'com.urbanairship.android:urbanairship-sdk:8.8.1'
```
Comment (BrianBatchelder) [inline src/android/build-extras.gradle]: 8.8.2 is now out. Go ahead and use it.

File: CHANGELOG [hunk 6]
```diff
@@ -1,4 +1,11 @@
 =============================
+Version 6.9.0 - August 10, 2017
+=============================
+- Updated iOS Urban Airship SDK to 8.5.2
+- Updated Android Urban Airship SDK to 8.8.1
```
Comment (BrianBatchelder) [inline CHANGELOG]: 8.8.2

File: www/UrbanAirship.js (hunk 1 above)
Comment (crow) [inline www/UrbanAirship.js]: added

File: src/android/UAirshipPlugin.java (hunk 2 above)
Comment (crow) [inline src/android/UAirshipPlugin.java]: nope

File: src/android/UAirshipPlugin.java (hunk 3 above)
Comment (crow) [inline src/android/UAirshipPlugin.java]: I cut and pasted to reformat it

Comment (crow) [inline src/android/UAirshipPlugin.java]: I cut and paste from the sample into the source directory - it looks like xcode did its reformatting magic.

File: src/android/UAirshipPluginManager.java [hunk 7]
```diff
@@ -102,6 +112,30 @@ void deepLinkReceived(String deepLink) {
...
     /**
+     * Called on app resume and when registration changes.
+     *
+     * @param context The application context.
+     */
+    void checkOptInStatus(Context context) {
+        boolean optIn = UAirship.shared().getPushManager().isOptIn();
+
+        if (preferences == null) {
+            preferences = context.getSharedPreferences(OPT_IN_STATUS_SHARED_PREFERENCES_FILE, Context.MODE_PRIVATE);
```
Comment (rlepinski) [inline src/android/UAirshipPluginManager.java]: can we make this a more general shared preferences file? We will most likely need to use it in the future for somethign else

**Comment by rlepinski** [comment]
👍  after minor change

**Comment by crow** [comment]
now:

private static final String UA_PLUGIN_SHARED_PREFERENCES_FILE = "com.urbanairship.ua_plugin_shared_prefereces";

**Comment by rlepinski** [comment]
👍

**Pull request merged**